src/main/java/in/ashwanthkumar/gocd/github/provider/gitlab/GitLabProvider.java

```diff
@@ -0,0 +1,158 @@
+package in.ashwanthkumar.gocd.github.provider.gitlab;
+
+import com.thoughtworks.go.plugin.api.GoPluginIdentifier;
+import com.tw.go.plugin.model.GitConfig;
+import com.tw.go.plugin.util.StringUtil;
+import in.ashwanthkumar.gocd.github.provider.Provider;
+import in.ashwanthkumar.gocd.github.provider.github.GHUtils;
+import in.ashwanthkumar.gocd.github.provider.github.model.PullRequestStatus;
+import in.ashwanthkumar.gocd.github.settings.general.DefaultGeneralPluginConfigurationView;
+import in.ashwanthkumar.gocd.github.settings.general.GeneralPluginConfigurationView;
+import in.ashwanthkumar.gocd.github.settings.scm.DefaultScmPluginConfigurationView;
+import in.ashwanthkumar.gocd.github.settings.scm.ScmPluginConfigurationView;
+import in.ashwanthkumar.gocd.github.util.URLUtils;
+import in.ashwanthkumar.utils.func.Function;
+import in.ashwanthkumar.utils.lang.StringUtils;
+import org.gitlab4j.api.GitLabApi;
+import org.gitlab4j.api.GitLabApiException;
+import org.gitlab4j.api.models.Author;
+import org.gitlab4j.api.models.MergeRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Properties;
+
+public class GitLabProvider implements Provider {
+    private static final Logger LOG = LoggerFactory.getLogger(GitLabProvider.class);
+    public static final String REF_SPEC = "+refs/merge-requests/*/head:refs/remotes/origin/merge-requests/*";
+    public static final String REF_PATTERN = "refs/remotes/origin/merge-requests/";
+    private static String LOGIN = "login";
+    private static String PASSWORD = "password";
+
+    @Override
+    public GoPluginIdentifier getPluginId() {
+        return new GoPluginIdentifier("gitlab.pr", Arrays.asList("1.0"));
+    }
+
+    @Override
+    public String getName() {
+        return "Gitlab";
+    }
+
+    @Override
+    public void addConfigData(GitConfig gitConfig) {
+        try {
+            Properties props = GitLabUtils.readPropertyFile();
+            if (StringUtil.isEmpty(gitConfig.getUsername())) {
+                gitConfig.setUsername(props.getProperty(LOGIN));
+            }
+            if (StringUtil.isEmpty(gitConfig.getPassword())) {
+                gitConfig.setPassword(props.getProperty(PASSWORD));
+            }
```

Comment:
Can you also please update the README with the format of the config file and how to setup the plugin?